src/Mvc/Mvc.ViewFeatures/src/RazorComponents/IComponentRenderer.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.RazorComponents
+{
+    internal interface IComponentRenderer
```

Comment:
What do we get out of this? If we just called the HTML helper from the tag helper, what are the problmes?

Reply:
Assuming we at least add the non-generic overload to the HTML helper's existing method right?

Reply:
Having the things as extensions makes it really difficult to test. Also the extension method is just a place to dump code. Outside of using the `ViewContext`, none of the `IHtmlHelper` is used.

Reply:
Can we at least get rid of the interface and use an abstract class instead?